docs/source/requests.md

```diff
@@ -985,6 +985,209 @@ Command to change Pool's configuration
 }
 ```
 
+### AUTH_RULE
+
+Command to change authentication rules. 
+Authentication rules are stored as key - value dictionary.
+Key - some action in the format `action--txn_type--field--old_value--new_value`
+Value is a set of constraints on the execution of this action. There are two types of constraints:
+- ConstraintEntity contains `{constraint_id, role, sig_count, need_to_be_owner, metadata}`
+- ConstraintList with format `{constraint_id, auth_constraints}` contains list of constraints
+That is, the entry 
+```
+"EDIT--NODE--services--[VALIDATOR]--[]" -> {constraint_id: OR,
+                                            auth_constraints: [{constraint_id: ROLE,
+                                                                role: STEWARD, 
+                                                                sig_count: 1, 
+                                                                need_to_be_owner: True},
+                                                               {constraint_id: ROLE,
+                                                                role: TRUSTEE, 
+                                                                sig_count: 1, 
+                                                                need_to_be_owner: False}
+                                                               ]
+                                           }
+                                                                 
+```
+means that change the value of node services from [VALIDATOR] to [] (demotion of node) can only TRUSTEE or STEWARD if it is owner of this transaction.
+
+- `auth_action` (enum: `ADD` or `EDIT`):
+
+    Change the rights to an action.
+    
+- `auth_type` (string):
+
+    The type of transaction to change rights to. (Example: NYM, NODE, POOL_RESTART...)
+
+- `field` (string):
+
+    Change the rights to edit(add) some values from field.
+
+- `old_value` (string; optional):
+
+    Old value of field, which can be changed to a new_value.
+
+- `new_value` (string):
+   
+   New value that can be used to fill the field.
+
+**ConstraintType:**
+
+ConstraintList
+
+- `constraint_id` (enum: `AND` or `OR`):
+
+    Type of a constraint class. Needed for determine the type of constraint for correct deserialization.
+    - `AND` logical conjunction for all constraints from `auth_constraints`
+    - `OR` logical disjunction for all constraints from `auth_constraints`
+    
+- `auth_constraints` (list of ConstraintType):
+
+    List of ConstraintType (ConstraintList or ConstraintEntity) objects
+    
+ ```
+{ 'constraint_id': 'AND',
+  'auth_constraints': [<ConstraintEntity>,
+                      <ConstraintEntity>]
+}
+```
+    
+ConstraintEntity
+
+- `constraint_id` (enum: `ROLE`):
+
+   Type of constraint, which for ConstraintEntity is always of type "ROLE". Needed for determine the type of constraint for correct deserialization.
```

Comment:
Type of a constraint. As of not only `ROLE` is supported, but plugins can register new ones.